src/app/modules/twilio/twilio.service.ts

```diff
@@ -0,0 +1,25 @@
+import { ITwilioSmsWebhookBody } from 'src/types/twilio'
+
+import { createLoggerWithLabel } from '../../config/logger'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ *
+ * @param body body from Twilio Webhook with the failed sms details
+ */
+export const logFailedSmsDelivery = (body: ITwilioSmsWebhookBody): void => {
+  const { To, MessageSid, MessageStatus, ErrorCode, ErrorMessage } = body
+
+  logger.error({
+    message: 'Error occurred when attempting to send SMS on twillio',
+    meta: {
+      action: 'sendingTwilioSmsFailed',
```

Comment:
`meta.action` seems to be different from the service name. We should keep it congruent